src/components/axes/timeAxis.ts

```diff
@@ -207,6 +202,19 @@ export module Axis {
         return this._possibleTimeAxisConfigurations;
       }
       this._possibleTimeAxisConfigurations = configurations;
+      this._numTiers = _Util.Methods.max(this._possibleTimeAxisConfigurations.map((config: TimeAxisConfiguration) => config.length), 0);
+
+      if (this._isAnchored) {
+        this._createDomElements();
+      }
+
+      var oldLabelPositions: string[] = this.tierLabelPositions();
+      var newLabelPositions: string[] = [];
+      for (var i = 0; i < this._numTiers; i++) {
+        newLabelPositions.push(oldLabelPositions[i] || "between");
+      }
+      this.tierLabelPositions(newLabelPositions);
```

Comment:
Do we want the tierLabelPositions instance variable to always contain complete valid information (inputting "between" where needed)?

Or do we want to instead fill in the gaps in the instance variable when we are using tierLabelPositions?

The good thing that you've done here is that you've made the positions valid at every step.

The bad thing is that now this setter is really heavy, doing more than possibly the user expects.